internal/engine/eval/vulncheck/pkgdb.go

```diff
@@ -51,11 +51,13 @@ func urlFromEndpointAndPaths(
 type patchLocatorFormatter interface {
 	LineHasDependency(line string) bool
 	IndentedString(indent int, oldDepLine string, oldDep *pb.Dependency) string
+	HasPatchedVersion() bool
 }
 
 // RepoQuerier is the interface for querying a repository
 type RepoQuerier interface {
-	SendRecvRequest(ctx context.Context, dep *pb.Dependency) (patchLocatorFormatter, error)
+	SendRecvRequest(ctx context.Context, dep *pb.Dependency, patched string, latest bool) (patchLocatorFormatter, error)
+	NoPatchAvailableFormatter(dep *pb.Dependency) patchLocatorFormatter
```

Comment:
Do we actually use the `NoPatchAvailableFormatter` directly under any condition or do we display that we can't find a fix version and we use this formatter to fullfill a contract? (btw the formatter interface has outlived its lifespan I think and is ripe for refactoring (see also issues #1203 and #1243)

Reply:
We use the `LineHasDependency` method of the formatter to find which line to comment on, even if the comment says there's not fix available